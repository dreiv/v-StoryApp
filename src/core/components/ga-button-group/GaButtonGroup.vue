<script setup lang="ts">
import { provide } from 'vue'
import { uniqueId } from '@/core/utils/uniqueId'
import { buttonGroupKey } from './types'

export interface ButtonGroupProps {
  name?: string
}

const { name = uniqueId('group') } = defineProps<ButtonGroupProps>()
const model = defineModel<string>({ default: '' })

provide(buttonGroupKey, { name, model })
</script>

<template>
  <fieldset :class="$style.group">
    <slot />
  </fieldset>
</template>

<style module>
.group {
  display: inline-flex;
  flex-wrap: wrap;
  gap: var(--ga-size-spacing-01);

  border: var(--ga-size-border-width-sm) solid var(--ga-color-border-primary);
  border-radius: var(--ga-radius);
  padding: var(--ga-size-spacing-01);
}
</style>
