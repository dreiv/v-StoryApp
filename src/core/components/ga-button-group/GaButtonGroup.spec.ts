import { shallowMount, type VueWrapper } from '@vue/test-utils'
import GaButtonGroup from './GaButtonGroup.vue'

vi.mock('./types', () => ({ buttonGroupKey: 'buttonGroupKey' }))

const buttonGroup = { name: 'mockName', model: { value: '' } }
describe('ButtonGroup', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount(GaButtonGroup, { global: { provide: { buttonGroup } } })
  })

  it('should render correctly with default props', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render children in the default slot', () => {
    wrapper = shallowMount(GaButtonGroup, {
      slots: { default: '<div class="test-button">Button</div>' },
    })
    expect(wrapper.find('.test-button').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
