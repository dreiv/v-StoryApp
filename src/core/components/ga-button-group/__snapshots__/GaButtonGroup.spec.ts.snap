// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonGroup > should render children in the default slot 1`] = `
"<fieldset class="group">
  <div class="test-button">Button</div>
</fieldset>"
`;

exports[`ButtonGroup > should render correctly with default props 1`] = `"<fieldset class="group"></fieldset>"`;
